### PR TITLE
Balder/reduce jdisc http httprequest infestation

### PR DESCRIPTION
**Pull request by baldersheim**
@bjorncs or @bratseth PR

### CODE REVIEW

**Review by bratseth — APPROVED** [review]
Nice

File: container-search/src/main/java/com/yahoo/search/Query.java [hunk 1]
```diff
@@ -912,7 +913,9 @@ private void copyPropertiesTo(Query clone) {
      * Return the HTTP request which caused this query. This will never be null
      * when running with queries from the network.
      */
-     public HttpRequest getHttpRequest() { return httpRequest; }
+    public HttpRequest getHttpRequest() { return httpRequest; }
+
+    public URI getUri() { return httpRequest != null ? httpRequest.getUri() : null; }
```
Comment (bjorncs) [inline container-search/src/main/java/com/yahoo/search/Query.java:918]: When will `httpRequest` be null? Should the return value be `Optional<URI>`?

**Review by bjorncs — COMMENTED** [review]
👍

File: container-search/src/main/java/com/yahoo/search/Query.java (hunk 1 above)
Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/search/Query.java:918]: Currently it can not be null.

Comment (bjorncs) [reply container-search/src/main/java/com/yahoo/search/Query.java:918]: So that will be changed at a later time?

Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/search/Query.java:918]: Hard to tell. If a query, fx created for testing, makes sense without URI we should make it Optional.

**Pull request merged**